src/clojure/clojang/jinterface/otp/messaging.clj

```diff
@@ -108,16 +168,17 @@
   (whereis [this mbox-name]
     "Determine the pid corresponding to a registered name on this node."))
 
-(extend-type OtpMbox MboxObject
+(extend-type OtpMbox
+  MboxObject
   (close [this]
     (.close this))
   (equal? [this other-obj]
     (.equals this other-obj))
   (exit
     ([this reason]
-      (.exit this))
-    ([this recip-pid reason]
-      (.exit recip-pid reason)))
+     (.exit this ^OtpErlangObject reason))
+    ([this ^OtpErlangPid recip-pid ^OtpErlangObject reason]
```

Comment:
This should optionally allow a `String` `node-name` and `String` `reason`, but for now `^OtpErlangPid` and `^OtpErlangObject` are hard-coded, respectively.